packages/react-core/src/components/Label/Label.tsx

```diff
@@ -254,7 +257,9 @@ export const Label: React.FunctionComponent<LabelProps> = ({
   };
 
   let labelComponentChild = (
-    <LabelComponentChildElement {...labelComponentChildProps}>{content}</LabelComponentChildElement>
+    <LabelComponentChildElement onClick={onLabelClick} {...labelComponentChildProps}>
```

Comment:
this is going to need `type="button"` if it's a `<button>`, since a button's default `type` is "submit", which will submit a form if it's placed in a form and you click on it.